src/components/common/Typography/Form/CollapsibleArrayTemplate/CollapsibleArrayTemplate.jsx

```diff
@@ -17,71 +18,97 @@ const renderLabel = (label, displayLabel) => {
 /**
  * This is collapsible array template for array field.
  */
-const CollapsibleArrayTemplate = (props) => {
-  const [indexToOpen, changeIndex] = useState(undefined);
-  const {
-    id,
-    classNames,
-    label,
-    displayLabel,
-    errors,
-    items,
-    uiSchema,
-    canAdd,
-    title,
-  } = props;
-
-  const showErrors = uiSchema && uiSchema['ui:options'] ? !!uiSchema['ui:options'].showErrors : true;
-
-  // For rendering the errors to the field.
-  const renderErrors = () => {
-    if (showErrors && id !== 'root') {
-      return (
-        <div className="form-field-errors">
-          {errors}
-        </div>
-      );
-    }
-    return null;
-  };
+class CollapsibleArrayTemplate extends React.Component {
+  constructor(props) {
+    super(props);
 
-  const renderTitle = () => {
-    if (title && title.length) {
-      return <p className="collapsible-array-template__form-expander-heading">{title}</p>;
-    }
-    return null;
-  };
-
-  const onClickEdit = (index, isClicked) => {
-    if (isClicked) {
-      changeIndex(index);
-    } else {
-      changeIndex(undefined);
+    this.state = {
+      indexToOpen: undefined,
+    };
+  }
+
+  componentDidUpdate(prevProps) {
+    // This code will automatically opens the newly add item
+    if ((prevProps.formData).length < (this.props.formData).length) {
+      if ((prevProps.formData).length === 1 && (isEmpty(prevProps.formData[0]) || !prevProps.formData[0].name)) {
+        this.changeIndex(undefined);
+      } else {
+        this.changeIndex((this.props.items).length - 1);
+      }
     }
-  };
-
-  return (
-    <div
-      id={id}
-      className={classNames}
-    >
-      {renderTitle()}
-      {renderLabel(label, displayLabel)}
-      <ArrayTemplateAddButton canAdd={canAdd} onAddClick={props.onAddClick} />
-      {items && items.map((element, index) => (
-        <ArrayTemplateForm
-          element={element}
-          isEditClicked={indexToOpen === index}
-          index={index}
-          onClickEdit={onClickEdit}
-          haveError={props.uiSchema['ui:options'].haveError}
-        />
+  }
+
+  changeIndex = (index) => {
+    this.setState({
+      indexToOpen: index,
+    });
+  }
+
+  render() {
+    const {
+      id,
+      classNames,
+      label,
+      displayLabel,
+      errors,
+      items,
+      uiSchema,
+      canAdd,
+      title,
+    } = this.props;
+
+    const showErrors = uiSchema && uiSchema['ui:options'] ? !!uiSchema['ui:options'].showErrors : true;
+
+    // For rendering the errors to the field.
+    const renderErrors = () => {
+      if (showErrors && id !== 'root') {
+        return (
+          <div className="form-field-errors">
+            {errors}
+          </div>
+        );
+      }
+      return null;
+    };
+
+    const renderTitle = () => {
+      if (title && title.length) {
+        return <p className="collapsible-array-template__form-expander-heading">{title}</p>;
```

Comment:
Use Paragraph component
```suggestion
        return <Paragraph className="collapsible-array-template__form-expander-heading">{title}</p>;
```

Reply:
Done